### PR TITLE
Terraform upgrade 0.12

### PR DESCRIPTION
**Pull request by MalloZup**
# What does this PR:

UPGRADE to TERRAFORM V12:

- encapsulate PR: https://github.com/dmacvicar/terraform-provider-libvirt/pull/602 from @tripledes
-  fix testACC failures due to new syntax

## TestACC testuite:

- [X] green

## Documentation:

- [x] documentation need to be adapted because I think we need change some examples with New Terraform syntax.

### CODE REVIEW

**Comment by MalloZup** [comment]
regarding this `TestAccLibvirtDomain_KernelInitrdCmdline`, I didn't had the time to look at. If anyone want to jump on this feel free ! :rocket:

**Comment by MalloZup** [comment]
I have updated the documentation which got changed.
Since our tests use already our API I assume that everything that I didn't need to change works already.

It might be that we have some corner cases where the documentation need some update but from my pov we can do later after release. Since the testacc is green

**Comment by tripledes** [comment]
@MalloZup I've done some [doc updates](https://github.com/tripledes/terraform-provider-libvirt/commit/fe2a2a4e032ae06320b8aa297c411afd79459d06) myself. Let me know if you want me to create a PR against this branch or you cherry pick the commit yourself :beers:

**Comment by MalloZup** [comment]
Thx! Feel free to submit a pr against this branch and I will merge then

**Comment by MalloZup** [comment]
@dmacvicar to me we can merge this and provide the `devel`/head 0.12 terraform and the release version (pkg, etc) can follow-up as always we did in past.

**Comment by thundertaker** [comment]
Do I understand correctly that the only step left is code review for this branch, and then it will be merged to master (and then release, packages, etc. described in #605)?

**Comment by MalloZup** [comment]
I"m going to merge this for enable people to use TF0.12 already. As I stated the release can come later on . I left 10 days PR open in case Duncan wanted to review. Since is not a Critical design change on the code base, but just an update we can merge it. 

thx @tripledes

File: libvirt/resource_libvirt_domain_test.go [hunk 1]
```diff
@@ -412,13 +412,18 @@ func TestAccLibvirtDomain_KernelInitrdCmdline(t *testing.T) {
...
+
+
+		cmdline = [ 
+			{
+			arg1 = "value1"
+			}
+		 ]
+		cmdline = [
+		 {
+			arg1 = "value2"
```
Comment (zeenix) [inline libvirt/resource_libvirt_domain_test.go]: @MalloZup did you run this testcase after the change? It fails here:

```
=== RUN   TestAccLibvirtDomain_KernelInitrdCmdline
--- FAIL: TestAccLibvirtDomain_KernelInitrdCmdline (0.00s)
    testing.go:568: Step 0 error: /tmp/tf-test083582606/main.tf:27,3-10: Attribute redefined; The argument "cmdline" was already set at /tmp/tf-test083582606/main.tf:22,3-10. Each argument may be set only once.
FAIL
```

Comment (zeenix) [reply libvirt/resource_libvirt_domain_test.go]: It's OK. Fix is coming as part of #600. Once that's merged, we can avoid breakage like this. :)

Comment (MalloZup) [reply libvirt/resource_libvirt_domain_test.go]: @zeenix yep ! thx  I will submit a PR for this

Comment (MalloZup) [reply libvirt/resource_libvirt_domain_test.go]: I think i forgot to push changes.. Anyways i'm testing the PR with testacc now locally.. THx for spotting it! :rocket:

Comment (MalloZup) [reply libvirt/resource_libvirt_domain_test.go]: https://github.com/dmacvicar/terraform-provider-libvirt/pull/611/files
